gitlabform/processors/group/group_ldap_links_processor.py

```diff
@@ -1,18 +1,63 @@
+from logging import debug
+from typing import List
+
 from gitlabform.gitlab import GitLab
-from gitlabform.processors.defining_keys import And, Or, Key, Xor
-from gitlabform.processors.multiple_entities_processor import MultipleEntitiesProcessor
+from gitlab.v4.objects import Group
+from gitlabform.processors.abstract_processor import AbstractProcessor
+
 
+class GroupLDAPLinksProcessor(AbstractProcessor):
 
-class GroupLDAPLinksProcessor(MultipleEntitiesProcessor):
     def __init__(self, gitlab: GitLab):
-        super().__init__(
-            "group_ldap_links",
-            gitlab,
-            list_method_name="get_ldap_group_links",
-            add_method_name="add_ldap_group_link",
-            delete_method_name="delete_ldap_group_link",
-            defining=And(Key("provider"), Or(Key("cn"), Key("filter"))),
-            required_to_create_or_update=And(
-                Key("provider"), Xor(Key("cn"), Key("filter"))
-            ),
-        )
+        super().__init__("group_ldap_links", gitlab)
+
+    def _process_configuration(self, group_path: str, configuration: dict):
+
+        group: Group = self.gl.get_group_by_path_cached(group_path)
+
+        configured_links = configuration["group_ldap_links"]
+        enforce_links = configuration.get("group_ldap_links|enforce", False)
+
+        existing_links: List[dict] = self._fetch_ldap_links(group)
+
+        for name, link_config in configured_links.items():
+            if name == "enforce":
+                continue
+
+            self._update_ldap_link(group, existing_links, link_config)
+
+        if enforce_links:
+            self._delete_extra_links(group, existing_links, configured_links)
+
+    def _fetch_ldap_links(self, group: Group) -> List[dict]:
+        links = group.ldapgrouplinks.list(all=True)
+        return [link.attributes for link in links]
```

Comment:
why not return the full Link Object in the list rather than the attributes, then later you can call link.get_id() etc, rather than link["id"] it's more robust and easier to read